### ISSUE TITLE
ERROR: MKLDNNGraph::CreateGraph: such topology cannot be compiled for dynamic batch!

### ISSUE THREAD

**Issue by Semen52** [opencv/dldt#30]
Hi guys,

I'm trying to use SSD V1 in batch mode and got the following error:
ERROR: MKLDNNGraph::CreateGraph: such topology cannot be compiled for dynamic batch!

Is it possible?

**Comment by ilyachur** [comment]
Hi @Semen52,

Yes, it is possible. 

Currently, dynamic batch is not supported for all topologies by the CPU plugin because the CPU plugin doesn't use shapeInfer() method and cannot detect right dimensions for some layers (like PriorBox, Permute, Reshape and etc).

**Comment by Semen52** [comment]
Any suggestions?

**Comment by Semen52** [comment]
As I understand the problem can be solved only after MKLDNNPlugin library update.

**Comment by ilyachur** [comment]
Yes the problem could be resolved only after fixes on the MKLDNNPlugin side.

You can try to create ExecutionNetwork with the dynamic batch support and disable dynamic batch if you get an exception.

**Comment by Semen52** [comment]
Thank you.

**Comment by elvin-nnov** [comment]
Semen, general question - do you need dynamically changed batch between inference calls or just run SSD in with batch size > 1? The second is doable right now using shape inference before loading of network to plugin like this
```
SizeVector input_shape;
std::tie(input_name, input_shape) = *input_shapes.begin();
input_shape[0] = batch_size;
input_shapes[input_name] = input_shape;
network.reshape(input_shapes);
ExecutableNetwork executable_network = plugin.LoadNetwork(network, {});
```

 if you want change batch size between inference calls - SSD network is not supported in this mode, this should be fixed in MKLDNNPlugin

**Comment by Semen52** [comment]
Hi @elvin-nnov,

I need the both feature =)
I'm using C++ API and before loading I've set a max batch size:
`network.setBatchSize(10);`

After that load a network using the following config:
`{ { InferenceEngine::PluginConfigParams::KEY_DYN_BATCH_ENABLED,
                InferenceEngine::PluginConfigParams::YES } }`

In inference I set a max batch size depends on input shape:
`network_infer_request.SetBatch((int)current_batch_size);`

For now, the above scenario isn't supported.
I guess two scenarios are available:
- Convert a model using key --batch. Disable dynamic batch. Run model.
- Reshape input (is it supported in C++ API?). Disable dynamic batch. Run model.
